Sources/PackagePlugin/Plugin.swift

```diff
@@ -104,27 +104,36 @@ extension Plugin {
         }
     }
     
+    /// Handles a single message received from the plugin host.
     fileprivate static func handleMessage(_ message: HostToPluginMessage) async throws {
         switch message {
-
```

Comment:
The section that starts here is one of the key changes, and moves the plugin action that was nested inside `performAction` to instead be up at the top level.  The simplifies the structure and has the plugin message contain the input structure from which the package context is formed.